dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -560,38 +562,53 @@ public BlockReader createFileReader(String fileId, long offset, boolean position
   }
 
   @Override
-  public ListenableFuture<List<FileFailure>> load(List<File> files, UfsReadOptions options) {
+  public ListenableFuture<List<LoadFileFailure>> load(List<UfsStatus> ufsStatuses, UfsReadOptions options) {
+    System.out.println("Request: " + Arrays.toString(ufsStatuses.toArray()));
     List<ListenableFuture<Void>> futures = new ArrayList<>();
-    List<FileFailure> errors = Collections.synchronizedList(new ArrayList<>());
-    for (File file : files) {
-      ListenableFuture<Void> loadFuture = Futures.submit(() -> {
-        try {
-          if (options.hasUser()) {
-            AuthenticatedClientUser.set(options.getUser());
+    List<LoadFileFailure> errors = Collections.synchronizedList(new ArrayList<>());
+    for (UfsStatus status : ufsStatuses) {
+      String ufsFullPath = status.getUfsFullPath().toString();
+      DoraMeta.FileStatus fs = buildFileStatusFromUfsStatus(status, ufsFullPath);
+      mUfsStatusCache.put(ufsFullPath, fs);
+      if (mMetaStore != null) {
+        mMetaStore.putDoraMeta(ufsFullPath, fs);
```

Comment:
working on a PR to extract the dora metadata interface 
https://github.com/Alluxio/alluxio/pull/17458#pullrequestreview-1450108606

Reply:
shouldnt we put into cache + metastore after successful load? coz these ufs status comes from scheduler and might be a retried one and might be stale

Reply:
how can you load the data alone without having the metadata of the file?